node/pkg/db/governor.go

```diff
@@ -176,12 +190,35 @@ func unmarshalOldTransfer(data []byte) (*Transfer, error) {
 	}
 	t.EmitterAddress = emitterAddress
 
-	msgID := make([]byte, 256)
-	n, err := reader.Read(msgID)
-	if err != nil || n == 0 {
-		return nil, fmt.Errorf("failed to read vaa id [%d]: %w", n, err)
+	msgIdLen := uint16(0)
+	if err := binary.Read(reader, binary.BigEndian, &msgIdLen); err != nil {
+		return nil, fmt.Errorf("failed to read msgID length: %w", err)
 	}
-	t.MsgID = string(msgID[:n])
+
+	if msgIdLen > 0 {
+		msgID := make([]byte, msgIdLen)
+		n, err := reader.Read(msgID)
+		if err != nil || n == 0 {
+			return nil, fmt.Errorf("failed to read vaa id [%d]: %w", n, err)
+		}
+		t.MsgID = string(msgID[:n])
+	}
+
```

Comment:
Do you care if t.MsgID doesn't have a value?

Reply:
I don't think so.